runtime/lua/vim/lsp/buf.lua

```diff
@@ -839,6 +840,18 @@ function M.code_action(options)
       params = util.make_range_params(win, client.offset_encoding)
     end
     params.context = context
+
+    -- Don't send markup to clients that don't support it.
+    if not client.server_capabilities.textDocument.diagnostic.markupMessageSupport then
+      --- @param diagnostic lsp.Diagnostic
+      context.diagnostics = vim.iter.map(function(diagnostic)
+        if type(diagnostic.message) == 'table' then
+          diagnostic.message = diagnostic.message.value
+        end
+        return diagnostic
+      end, context.diagnostics)
```

Comment:
Suggest simplifying this to a simple for loop

```suggestion
      for _, v in ipairs(context.diagnostics)
        if type(v.message) == 'table' then
          v.message = v.message.value
        end
      end
```